web-common/src/components/searchable-filter-menu/SearchableFilterDropdown.svelte

```diff
@@ -0,0 +1,111 @@
+<script lang="ts">
+  import Check from "../icons/Check.svelte";
+  import Spacer from "../icons/Spacer.svelte";
+  import { Menu, MenuItem } from "../menu";
+  import { Search } from "../search";
+  import Footer from "./Footer.svelte";
+  import Button from "../button/Button.svelte";
+  import { createEventDispatcher } from "svelte";
+  import { matchSorter } from "match-sorter";
+
+  const dispatch = createEventDispatcher();
+
+  export let selectableItems: string[];
+  export let selectedItems: boolean[];
+
+  type MenuItemData = {
+    label: string;
+    selected: boolean;
+    index: number;
+  };
+
+  export const setItemsVisibleBySearchString = (
+    items: string[],
+    selected: boolean[],
+    searchText: string
+  ): MenuItemData[] => {
+    let menuEntries = items.map((item, i) => ({
+      label: item,
+      selected: selected[i],
+      index: i,
+    }));
+    return matchSorter(menuEntries, searchText, { keys: ["label"] });
+  };
+
+  let searchText = "";
+
+  $: menuItems = setItemsVisibleBySearchString(
+    selectableItems,
+    selectedItems,
+    searchText
+  );
+
+  $: numSelected = selectedItems?.filter((s) => s)?.length || 0;
+
+  $: numSelectedNotShown =
+    numSelected - (menuItems?.filter((item) => item.selected)?.length || 0);
+
+  $: allToggleText = numSelected === 0 ? "Select all" : "Deselect all";
+
+  $: allToggleEvt = numSelected === 0 ? "selectAll" : "deselectAll";
```

Comment:
I think we're not 100% consistent here, but I tend to prefer `select-all` kebab casing for component events, and this is what's most dominant in the code I think.